datafusion/sql/src/planner.rs

```diff
@@ -2711,6 +2711,15 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_int_decimal_default() {
+        quick_test(
+            "SELECT CAST(10 AS DECIMAL)",
```

Comment:
Can we please add a test for the other cases?

* Specify only scale, use default precision: `CAST (10 AS DECIMAL(5))`
* Error that scale is larger than precision: `CAST(10 AS DECIMAL(5, 10))`